code/cheats_table/cheats_table.h

```diff
@@ -0,0 +1,124 @@
+#include "globalincs/globals.h"
+#include "globalincs/pstypes.h"
+#include "globalincs/systemvars.h"
+#include "io/key.h"
+#include "hud/hudmessage.h"
+#include "hud/hudsquadmsg.h"
+#include "scripting/global_hooks.h"
+#include "scripting/hook_api.h"
+#include "ship/ship.h"
+#include "ship/shipfx.h"
+
+#define CHEAT_BUFFER_LEN	17
+
+class CustomCheat {
+	public:
+		SCP_string cheatCode;
+		SCP_string cheatMsg;
+		bool requireCheatsEnabled;
+
+		CustomCheat(SCP_string cheat_code, SCP_string cheat_msg, bool require_cheats_enabled) {
+			cheatCode = cheat_code;
+			cheatMsg = cheat_msg;
+			requireCheatsEnabled = require_cheats_enabled;
+		}
+
+		virtual ~CustomCheat() = default;
+
+		virtual void runCheat() {
+			if (canUseCheat()) {
+				HUD_printf("%s", cheatMsg.c_str());
+				scripting::hooks::OnCheat->run(scripting::hook_param_list(scripting::hook_param("Cheat", 's', cheatCode)));
+				CheatUsed = SCP_string(cheatCode);
+			}
+		}
+
+		bool canUseCheat() {
+			return !requireCheatsEnabled || Cheats_enabled;
+		}
+};
+
+class SpawnShipCheat : public CustomCheat {
+	protected:
+	SCP_string shipClassName;
+	SCP_string shipName;
+	public:
+
+	SpawnShipCheat(SCP_string cheat_code, SCP_string cheat_msg,  bool require_cheats_enabled, SCP_string class_name, SCP_string ship_name) : CustomCheat(cheat_code, cheat_msg, require_cheats_enabled) {
+		shipClassName = class_name;
+		shipName = ship_name;
+	}
+
+	void runCheat() override {
```

Comment:
This method, and really, all methods of these classes, should be in the cheats_table.cpp and not the .h